src/core/stdcpp/typeinfo.d

```
@@ -0,0 +1,146 @@
+            final bool before(const type_info rhs) const;
+            final const(char)* name() const;
+          protected:
+            //type_info();
+          private:
+            //this(const type_info rhs);
+            //type_info operator=(const type_info rhs);
+        }
+
+        class bad_cast : core.stdcpp.exception.std.exception
+        {
+            this() nothrow { }
+            this(const bad_cast) nothrow { }
+            //bad_cast operator=(const bad_cast) nothrow { return this; }
+            //virtual ~this() nothrow;
+            override const(char)* what() const nothrow;
+        }
+
+        class bad_typeid : core.stdcpp.exception.std.exception
+        {
+            this() nothrow { }
+            this(const bad_typeid) nothrow { }
+            //bad_typeid operator=(const bad_typeid) nothrow { return this; }
+            //virtual ~this() nothrow;
+            override const (char)* what() const nothrow;
+        }
+    }
+}
+else version (CRuntime_Microsoft)
+{
+    import core.stdcpp.exception;
+
+    struct __type_info_node
+    {
+        void* _MemPtr;
+        __type_info_node* _Next;
+    }
+
+    extern __gshared __type_info_node __type_info_root_node;
+
+    extern (C++, std)
+    {
+        class type_info
+        {
+
+          public:
+            //virtual ~this();
+            void dtor() { }     // reserve slot in vtbl[]
+            //bool operator==(const type_info rhs) const;
+            //bool operator!=(const type_info rhs) const;
+            final bool before(const type_info rhs) const;
+            final const(char)* name(__type_info_node* p = &__type_info_root_node) const;
+
+          private:
+            void* pdata;
+            char _name[1];
+            //this(const type_info rhs);
+            //type_info operator=(const type_info rhs);
+        }
+
+        class bad_cast : core.stdcpp.exception.std.exception
+        {
+            this(const(char)* msg = "bad cast") { }
```

@IgorStepanov, it's better that it issues a linker error than silently failing. Calling an empty D constructor doesn't do the right thing.